source/Halibut/Transport/ServerCertificateInterceptor.cs

```diff
@@ -1,72 +1,108 @@
-#if SUPPORTS_WEB_SOCKET_CLIENT
 using System;
 using System.Collections.Generic;
 using System.Net;
+using System.Net.Security;
+using System.Net.WebSockets;
 using System.Security.Cryptography.X509Certificates;
 
 namespace Halibut.Transport
 {
-    static class ServerCertificateInterceptor
+    class ServerCertificateInterceptor
     {
         public const string Header = "X-Octopus-RequestId";
 
-        static readonly Dictionary<string, X509Certificate2> certificates = new();
-        static bool initialized;
+        readonly Dictionary<string, X509Certificate2> certificates = new();
+        bool initialized;
 
 
-        public static void Expect(string connectionId)
+        public void Expect(string connectionId, ClientWebSocket clientWebSocket)
         {
             lock (certificates)
             {
                 if (!initialized)
-                    Initialize();
+                    Initialize(clientWebSocket, connectionId);
 
                 certificates[connectionId] = null;
             }
         }
 
-        static void Initialize()
+        void Initialize(ClientWebSocket clientWebSocket, string connectionId)
         {
-            var previous = ServicePointManager.ServerCertificateValidationCallback;
-            ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, errors) =>
+#if SUPPORTS_WEB_SOCKET_CERTIFICATE_VALIDATION_CALLBACK
+            clientWebSocket.Options.RemoteCertificateValidationCallback = WebSocketServerCertificateValidationCallback(clientWebSocket.Options.RemoteCertificateValidationCallback, connectionId);
+#endif
+            ServicePointManager.ServerCertificateValidationCallback = ServerCertificateValidationCallback(ServicePointManager.ServerCertificateValidationCallback);
```

Comment:
Do we need both to be registered?

Reply:
I don't think so. I'll have a play and see!!